File: RedSheet.DbEntities/Models/ProjectOutcomeAndLearning.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectOutcomeAndLearnings", Schema = "dbo")]
	[Serializable]
	public partial class ProjectOutcomeAndLearning  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectOutcomeAndLearningId { get; set; } 
 
	[Required] 
	[MaxLength(400)]
	public string ProjectOutcomeAndLearningValue { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public OutcomeAndLearningCategory OutcomeAndLearningCategoryId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 

	public ProjectOutcomeAndLearning()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/project-outcome-and-learning.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule,  vProjectOutcomeAndLearningRecord  } from './'
export class ProjectOutcomeAndLearning {
    constructor(projectOutcomeAndLearning?: ProjectOutcomeAndLearning  | vProjectOutcomeAndLearningRecord )  {
        let properties = [ "projectOutcomeAndLearningId", "projectOutcomeAndLearningValue", "outcomeAndLearningCategoryId", "projectModuleId",];
        for (let property of properties)
            if (projectOutcomeAndLearning && projectOutcomeAndLearning[property])
                this[property] = projectOutcomeAndLearning[property];
    }
 
	projectOutcomeAndLearningId : number =   0 ;
 
    @required()
    @maxLength(400)
	projectOutcomeAndLearningValue : string =   undefined;
 
    @range(0,2147483647)
	outcomeAndLearningCategoryId : number =   undefined;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;


}
